src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/WebWorkerTest.cs

```diff
@@ -461,44 +463,87 @@ public async Task ManagedYield(Executor executor)
 
         private async Task ActionsInDifferentThreads<T>(Executor executor1, Executor executor2, Func<Task, TaskCompletionSource<T>, Task> e1Job, Func<T, Task> e2Job, CancellationTokenSource cts)
         {
-            TaskCompletionSource<T> readyTCS = new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously);
-            TaskCompletionSource doneTCS = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+            TaskCompletionSource<T> job1ReadyTCS = new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously);
+            TaskCompletionSource job2DoneTCS = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+            var e1Done = false;
+            var e2Done = false;
+            var e1Failed = false;
+            Task e1;
+            Task e2;
+            T r1;
+
+            async Task ActionsInDifferentThreads1()
+            {
+                try
+                {
+                    await e1Job(job2DoneTCS.Task, job1ReadyTCS);
+                    if (!job1ReadyTCS.Task.IsCompleted)
+                    {
+                        job1ReadyTCS.SetResult(default);
+                    }
+                    await job2DoneTCS.Task;
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("ActionsInDifferentThreads1 failed\n" + ex);
+                    job1ReadyTCS.SetResult(default);
+                    e1Failed = true;
+                    throw;
+                }
+                finally
+                {
+                    e1Done = true;
+                }
+            }
 
-            var e1 = executor1.Execute(async () =>
+            async Task ActionsInDifferentThreads2()
             {
-                await e1Job(doneTCS.Task, readyTCS);
-                if (!readyTCS.Task.IsCompleted)
+                try
                 {
-                    readyTCS.SetResult(default);
+                    await e2Job(r1);
```

Comment:
i'm surprised this works, since r1 is potentially uninitialized at this point

Reply:
note this is declaration of nested function, the execution order is below at lines 512 and 513